src/main/java/com/aws/greengrass/logmanager/CloudWatchAttemptLogsProcessor.java

```diff
@@ -399,6 +401,26 @@ private void updateCloudWatchAttemptLogInformation(String fileName,
         return new Pair(reachedMaxBatchSize, currBytesRead);
     }
 
+    /**
+     * Check if the log line is the start of a new log event.
+     *
+     * @param multiLineStartPattern        The pattern for start of new log event
+     * @param logLine                      The processed log line
+     * @return Whether the log line is the start of a new log event.
+     */
+    private boolean checkLogStartPattern(final Pattern multiLineStartPattern, final String logLine) {
+        if (multiLineStartPattern != null) {
+            try {
+                return multiLineStartPattern.matcher(logLine).find();
+            } catch (StackOverflowError e) {
+                logger.atWarn().cause(e).log("Unable to match log against pattern {}. Checking for trailing "
```

Comment:
you aren't checking for trailing whitespace though, you look at the beginning.

Reply:
yea type